core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1804,6 +1806,30 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def getAccessInstructions(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
+        val accessGroups = DBIO.sequence(workspaceContext.workspace.accessLevels.values.map { ref =>
```

Comment:
It's be nice if `DBIO` had `traverse` since `sequence (xs map f)` == `traverse(xs, f)`. But it doesn't, oh well